drivers/serial/uart_stm32.c

```diff
@@ -2146,6 +2149,10 @@ static int uart_stm32_registers_configure(const struct device *dev)
 			LL_EXTI_EnableIT_0_31(BIT(config->wakeup_line));
 		}
 	}
+#if defined(CONFIG_SOC_SERIES_STM32U5X) && HAS_LPUART
```

Comment:
I think this requires more thinking.
Put this way, feature is enabled by default as soon as a LPUART node is enabled. But this feature actually enables a dedicated clock, which will add consumption.
So I think this should remain an option with a way for user to select if active or not.

Some other considerations:
- Also available on H7 as CLKAM mode (and probably others as well, but since the name changes, it's not easy to track).
- Available on a range of peripherals instances (not all instances of a peripheral, so probably we need to track which peripheral supports it and if it is enabled).